mlflow/models/utils.py

```diff
@@ -821,6 +821,17 @@ def _enforce_mlflow_datatype(name, values: pd.Series, t: DataType):
     if is_upcast:
         return values.astype(numpy_type, errors="raise")
     else:
+        # support converting long -> float/double for 0 values
+        def all_zeros(xs):
+            return all(pd.isnull(x) or x == 0 for x in xs)
```

Comment:
What will be the value if the parameter is not provided?

Reply:
If the field is optional, then this field just becomes null when casting to pandas Series, and it should pass the check. Updated the test